source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -54,13 +64,49 @@ bool isTargetClusterLambdaGateway(Upstream::ClusterInfo const& cluster_info) {
   return egress_gateway_it->second.bool_value();
 }
 
+bool isContentTypeTextual(const Http::RequestOrResponseHeaderMap& headers) {
+  // If transfer-encoding is anything other than 'identity' (i.e. chunked, compress, deflate or
+  // gzip) then we want to base64-encode the response body regardless of the content-type value.
+  if (auto encoding_header = headers.TransferEncoding()) {
+    if (!absl::EqualsIgnoreCase(encoding_header->value().getStringView(),
+                                Http::Headers::get().TransferEncodingValues.Identity)) {
+      return false;
+    }
+  }
+
+  // If we don't know the content-type, then we can't make any assumptions.
+  if (!headers.ContentType()) {
+    return false;
+  }
+
+  const Http::LowerCaseString content_type_value{
+      std::string(headers.ContentType()->value().getStringView())};
+  if (content_type_value.get() == Http::Headers::get().ContentTypeValues.Json) {
+    return true;
+  }
+
+  if (content_type_value.get() == "application/javascript") {
+    return true;
+  }
+
+  if (content_type_value.get() == "application/xml") {
```

Comment:
I updated `application/json` to use the one in `HeaderValues`, however, these two content-types are not available in that class. LMK if you think it makes sense to add them that class.